packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/java/integrationsteps.md

```diff
@@ -42,53 +41,23 @@ mav.addObject(REDIRECT_URI,
 mav.addObject(ISSUER_URI, issuer);
 
 session.setAttribute(CODE_VERIFIER, idxClientContext.getCodeVerifier());
-
 ```
 
-> **Note:** In the previous code snippet used in the embedded Sign-In Widget sample app, the `REDIRECT_URI` is taken from the app start configuration setting (such as `https://localhost:8080`), as well as `http://localhost:8080/authorization-code/callback`.
-
 ### Step 3: Request tokens from Okta
 
-Use the interaction code and code verifier to request for tokens.
-
-```java
-public JsonNode exchangeCodeForToken(final String interactionCode, final String codeVerifier)
-      throws MalformedURLException {
-
-   final HttpHeaders headers = new HttpHeaders();
-   headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
-   headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
-
-   final MultiValueMap<String, String> requestParams = new LinkedMultiValueMap<>();
-   requestParams.add("grant_type", "interaction_code");
-   requestParams.add("client_id", clientId);
-   if (Strings.hasText(clientSecret)) {
-      requestParams.add("client_secret", clientSecret);
-   }
-   requestParams.add("interaction_code", interactionCode);
-   requestParams.add("code_verifier", codeVerifier);
-
-   final HttpEntity<MultiValueMap<String, String>> requestEntity =
-            new HttpEntity<>(requestParams, headers);
-   final String tokenUri = ClientUtil.getNormalizedUri(issuer, "/v1/token");
-
-   final ResponseEntity<JsonNode> responseEntity =
-            restTemplate.postForEntity(tokenUri, requestEntity, JsonNode.class);
-
-   return responseEntity.getBody();
-}
-```
+Use the interaction code and code verifier to request for tokens. See helper class [exchangeCodeForToken()](https://github.com/okta/okta-idx-java/blob/f9378d48d39c10c76294e079f35214bbef3a02cd/samples/embedded-sign-in-widget/src/main/java/com/okta/spring/example/HelperUtil.java#L80).
 
 ```java
 final JsonNode jsonNode = helperUtil.exchangeCodeForToken(interactionCode, codeVerifier);
 final OAuth2AccessToken oAuth2AccessToken = helperUtil.buildOAuth2AccessToken(jsonNode);
 final OAuth2RefreshToken oAuth2RefreshToken = helperUtil.buildOAuth2RefreshToken(jsonNode);
-
 ```
 
-### Step 4: Call user profile information (Optional)
+The Spring security framework doesn't understand Okta’s interaction code flow. Therefore, the app needs to intercept Spring’s OAuth authentication code flow, exchange the interaction code obtained from Okta for an access token, fill the user profile attributes, and construct [OAuth2AuthenticationToken.java](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java) to hand over authentication flow back to Spring.
```

Comment:
```suggestion
The Spring security framework does not understand Okta’s Interaction code flow. Therefore, your app will need to intercept Spring’s OAuth authentication code flow, exchange the interaction code obtained from Okta for an access token, fill the user profile attributes, and construct [OAuth2AuthenticationToken.java](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java) before handing over the authentication flow back to Spring.
```

Reply:
```suggestion
The Spring security framework doesn't understand Okta’s Interaction code flow. Therefore, your app needs to intercept Spring’s OAuth authentication code flow, exchange the interaction code obtained from Okta for an access token, populate the user profile attributes, and construct [OAuth2AuthenticationToken.java](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java) before handing over the authentication flow back to Spring.
```

Reply:
Sorry Arvind, I'm following our style guide convention to use contractions and present, active voice as much as possible.